### PR TITLE
Remove duplicated code

### PR DESCRIPTION
**Pull request by dominikl**
Remove the duplicated "Image to Zarr" code. Hope it doesn't mess something up. Just clashed with @manics 's PR #32.

### CODE REVIEW

**Comment by dominikl** [comment]
Of course...
```
src/omero_zarr/raw_pixels.py:74: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "str")
Found 1 error in 1 file (checked 6 source files)
```
What's that supposed to mean?

**Comment by manics** [comment]
I think it means the defined type of the variable is a string, but the expression returns a string or None.

**Comment by dominikl** [comment]
Ok. No idea how to fix that.

**Comment by joshmoore** [comment]
@dominikl: if `None` is ever an option, then you need to specify `Optional[str]` as the type so that users know that they need to perform a None check.

**Comment by dominikl** [comment]
Then I get `raw_pixels.py:50: error: Value of type variable "AnyStr" of "exists" cannot be "Optional[str]"` This is ridiculous...

**Comment by joshmoore** [comment]
Hope you can give it a chance. Once you have the annotations in place, it *is* valuable to know whether or not something can be `None`. Reading through something like https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html may give you a better feeling for what's going on.

In this case, on line https://github.com/ome/omero-cli-zarr/blob/4c348199c8def21d166742d3cc0b3471e8939f12/src/omero_zarr/raw_pixels.py#L47 you define `filename` without a `None` option. So the "compiler" assumes you want it to _always_ be non-`None`. Then in https://github.com/ome/omero-cli-zarr/blob/4c348199c8def21d166742d3cc0b3471e8939f12/src/omero_zarr/raw_pixels.py#L73-L78 you redefine it as a `str` *or* `None` so it's telling you something is odd.

Towards the top of the method, you can define:

```
filename: Optional[str] = None
```

to say that you are ok with the later definition. (Alternatively, if the blocks were in the reverse order it all would have "just worked" I think)

File: src/omero_zarr/raw_pixels.py [hunk 1]
```diff
@@ -11,89 +11,52 @@
 from zarr.hierarchy import Group, open_group
 
 
-def image_to_zarr(image: omero.gateway.ImageWrapper, args: argparse.Namespace) -> None:
-
-    cache_numpy = args.cache_numpy
+def image_to_zarr(image: omero.gateway.Image, args: argparse.Namespace) -> None:
```
Comment (manics) [inline src/omero_zarr/raw_pixels.py:14]:
```suggestion
def image_to_zarr(image: omero.gateway.ImageWrapper, args: argparse.Namespace) -> None:
```

File: src/omero_zarr/raw_pixels.py [hunk 2]
```diff
@@ -11,89 +11,52 @@
...
+    n_levels = add_image(image, root, cache_dir=cache_dir)
+    add_group_metadata(root, image, n_levels)
+    print("Finished.")
 
 
 def add_image(
-    image: omero.gateway.ImageWrapper, parent: Group, field_index: str = "0"
-) -> None:
-    """Adds the image pixel data as array to the given parent zarr group."""
+    image: omero.gateway.Image, parent: Group, cache_dir: Optional[str] = None
```
Comment (manics) [inline src/omero_zarr/raw_pixels.py:27]:
```suggestion
    image: omero.gateway.ImageWrapper, parent: Group, cache_dir: Optional[str] = None
```

**Comment by dominikl** [comment]
Thanks @manics . I thought I might have accidentally overwritten some of your changes in #32 .

**Review by manics — APPROVED** [review]

**Pull request merged**